actionpack/lib/action_controller/caching.rb

```diff
@@ -46,6 +46,20 @@ def cache_configured?
         end
     end
 
+    # Cache or yield the block. The cache is suppose to never expire.
+    #
+    # <tt>public</tt> By default the Cache-Control header is private, set this to
+    #   +true+ if you want your application to be cachable by other devices (proxy caches).
+    #
+    # <tt>version</tt> is the version passed as a key for the cache.
+    def http_cache_forever(public: false, version: 'v1')
+      expires_in 100.years, public: public
```

Comment:
Do any browsers or proxies have issues parsing dates this far in the future? e.g. http://en.wikipedia.org/wiki/Year_2038_problem